centreon/tests/e2e/features/Cloud-notifications/04-notification-edit/Index.ts

```diff
@@ -0,0 +1,266 @@
+    {
+      host: data.hosts.host1.name,
+      output: 'submit_status_2',
+      service: data.services.service1.name,
+      status: 'critical'
+    }
+  ]);
+
+  checkHostsAreMonitored([
+    {
+      name: data.hosts.host1.name,
+      status: 'down',
+      statusType: 'hard'
+    }
+  ]);
+
+  checkServicesAreMonitored([
+    {
+      name: data.services.service1.name,
+      status: 'critical'
+    }
+  ]);
+
+  waitUntilLogFileChange();
+});
+
+Then(
+  'only notifications for status changes of the updated resource parameters are sent',
+  () => {
+    if (!notificationEnabled) {
+      notificationSentCheck({
+        log: 'Notification on host 15',
+        contain: false
+      });
+      notificationSentCheck({
+        log: 'Notification on service (15,27)',
+        contain: false
+      });
+      return;
+    }
+
+    notificationSentCheck({ log: 'Notification on host 15' });
+
+    if (!notificationWithServices) {
+      notificationSentCheck({
+        log: 'Notification on service (15,27)',
+        contain: false
+      });
+      return;
+    }
+
+    notificationSentCheck({ log: 'Notification on service (15,27)' });
+  }
+);
+
+When('the user changes the contact configuration', () => {
+  cy.get('#Searchcontacts').click();
+  cy.wait('@getUsers');
+  cy.get('li > div > span.MuiButtonBase-root').eq(0).click();
+  cy.contains(contactAfterEdit).click();
+});
+
+When(
+  'the user selects the {string} action on a Notification Rule line',
+  (action) => {
+    cy.get('input.MuiSwitch-input').then(($checkbox) => {
+      switch (action) {
+        case 'enable':
+          // Firstly Deactivate the Notification
+          editNotificationBody.is_activated = false;
+          editNotification(editNotificationBody);
+          cy.reload();
+          cy.wait('@getNotifications');
+          // Reactivate the notification
+          cy.get('input.MuiSwitch-input').click().should('be.checked');
+          notificationEnabled = true;
+          break;
+        case 'disable':
+          cy.wrap($checkbox).click().should('not.be.checked');
+          notificationEnabled = false;
+          break;
+      }
+    });
+  }
+);
+
+Then(
+  'the notifications for status changes are sent only to the updated contact',
+  () => {
+    notificationSentCheck({
+      log: '[{"email_address":"admin@centreon.com","full_name":"admin admin"}]',
+      contain: false
+    });
+    notificationSentCheck({
+      log: '[{"email_address":"guest@localhost","full_name":"Guest"}]'
+    });
+  }
+);
+
+Then('{string} notification is sent for this rule once', (prefix) => {
+  if (!notificationEnabled) {
+    notificationSentCheck({
+      log: 'Notification on host 15',
+      contain: false
+    });
+    notificationSentCheck({
+      log: 'Notification on service (15,27)',
+      contain: false
+    });
+    return;
+  }
+
+  notificationSentCheck({ log: 'Notification on host 15' });
+  notificationSentCheck({ log: 'Notification on service (15,27)' });
```

Comment:
hostname and service description are not displayed in the logfile ? (instead of ids)

Reply:
@kduret this is an example of the logfile:

```
Mon Jan 29 10:45:30 2024: INFO: Time to check new configuration
Mon Jan 29 10:45:30 2024: INFO: Reading notification configuration:
Mon Jan 29 10:45:30 2024: ERROR: Unable to decode 'nil'. Is the URL still accessible?
Mon Jan 29 10:45:30 2024: INFO: Response code: 304
Mon Jan 29 10:45:30 2024: INFO: Service (15,27) --- state: 2 ; state_type: 1 ; last_check: 1706525109 ; last_hard_state: 1706525109 ; notif flags: 4
Mon Jan 29 10:45:30 2024: INFO: Notification on service (15,27) --- state: 2 ; notification_id: 2
Mon Jan 29 10:45:30 2024: INFO: {"notification_id":2,"channels":{"email":{"subject":"Subject notification for host group and services for these hosts and two contacts","formatted_message":"\u003Cp class=\u0022css-1qf631s-paragraph\u0022 dir=\u0022ltr\u0022\u003E\u003Cspan style=\u0022white-space: pre-wrap;\u0022\u003EBody notification for host group and services for these hosts and two contacts\u003C\/span\u003E\u003C\/p\u003E","contacts":[{"email_address":"contact1@localhost","full_name":"contact_1"},{"email_address":"contact2@localhost","full_name":"contact_2"}]},"slack":null,"sms":null}}
Mon Jan 29 10:45:30 2024: INFO: response code: 200
Mon Jan 29 10:45:30 2024: INFO: replacing {{LONGDATETIME}} by its value <<Monday January 29, 2024, 10:45:09>>
Mon Jan 29 10:45:30 2024: INFO: replacing {{STATE}} by its value <<CRITICAL>>
Mon Jan 29 10:45:30 2024: INFO: replacing {{NOTIFICATIONTYPE}} by its value <<PROBLEM>>
Mon Jan 29 10:45:30 2024: INFO: replacing {{NAME}} by its value <<host_1/service_1>>
Mon Jan 29 10:45:30 2024: INFO: replacing {{ALIAS}} by its value <<nil>>
Mon Jan 29 10:45:30 2024: INFO: replacing {{SHORTDATETIME}} by its value <<01/29/24 10:45:09>>
Mon Jan 29 10:45:30 2024: INFO: replacing {{ID}} by its value <<15:27>>
Mon Jan 29 10:45:30 2024: INFO: replacing {{OUTPUT}} by its value <<submit_status_2>>
Mon Jan 29 10:45:30 2024: INFO: replacing {{LONGDATETIME}} by its value <<Monday January 29, 2024, 10:45:09>>
Mon Jan 29 10:45:30 2024: INFO: replacing {{STATE}} by its value <<CRITICAL>>
Mon Jan 29 10:45:30 2024: INFO: replacing {{NOTIFICATIONTYPE}} by its value <<PROBLEM>>
Mon Jan 29 10:45:30 2024: INFO: replacing {{NAME}} by its value <<host_1/service_1>>
Mon Jan 29 10:45:30 2024: INFO: replacing {{ALIAS}} by its value <<nil>>
Mon Jan 29 10:45:30 2024: INFO: replacing {{SHORTDATETIME}} by its value <<01/29/24 10:45:09>>
Mon Jan 29 10:45:30 2024: INFO: replacing {{ID}} by its value <<15:27>>
Mon Jan 29 10:45:30 2024: INFO: replacing {{OUTPUT}} by its value <<submit_status_2>>
Mon Jan 29 10:45:30 2024: INFO: command content:
Mon Jan 29 10:45:30 2024: INFO: Sending notification -- output:
```

so basically we can use > <<host_1/service_1>>

Reply:
indeed, I think the test will be more robust with this